ledger-tool/src/main.rs

```diff
@@ -267,7 +268,12 @@ fn output_slot(
             "  Transactions: {}, hashes: {}, block_hash: {}",
             transactions, num_hashes, blockhash,
         );
-        println!("  Programs: {:?}", program_ids);
+        for (pubkey, count) in program_ids.iter() {
+            *all_program_ids.entry(*pubkey).or_insert(0) += count;
+        }
+        let mut program_ids_array: Vec<_> = program_ids.into_iter().collect();
+        program_ids_array.sort_by(|a, b| b.1.cmp(&a.1));
+        println!("  Programs: {:?}", program_ids_array);
```

Comment:
nit: Maybe subfunction for duplicate code?

Reply:
added